From 575d837fa27dcf51985821e011295324d73e8002 Mon Sep 17 00:00:00 2001
From: Tushar Nimkar <tnimkar@codeaurora.org>
Date: Sun, 11 May 2025 10:37:41 +0500
Subject: [PATCH] BACKPORT: drivers: lpm-levels: Return correct value while
 suspend Currently lpm_suspend_enter and lpm_cpuidle_s2idle return success
 always. This misleads on succcess and failure counts.

This change adds support to return correct value to caller.

Change-Id: I57dd0a4385f16f0e0aa1c92d602c0816ed358e2c
Signed-off-by: Tushar Nimkar <tnimkar@codeaurora.org>
---
 drivers/cpuidle/lpm-levels.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/cpuidle/lpm-levels.c b/drivers/cpuidle/lpm-levels.c
index dd9eb7e34822..b18f9cc75b5f 100644
--- a/drivers/cpuidle/lpm-levels.c
+++ b/drivers/cpuidle/lpm-levels.c
@@ -1521,6 +1521,7 @@ static void lpm_cpuidle_s2idle(struct cpuidle_device *dev,
 
 	cluster_unprepare(cpu->parent, cpumask, idx, false, 0, success);
 	cpu_unprepare(cpu, idx, true);
+	return ret;
 }
 
 #ifdef CONFIG_CPU_IDLE_MULTIPLE_DRIVERS
@@ -1741,7 +1742,7 @@ static int lpm_suspend_enter(suspend_state_t state)
 	}
 	if (idx < 0) {
 		pr_err("Failed suspend\n");
-		return 0;
+		return -EINVAL;
 	}
 
 	/*
@@ -1760,7 +1761,7 @@ static int lpm_suspend_enter(suspend_state_t state)
 
 	cluster_unprepare(cluster, cpumask, idx, false, 0, success);
 	cpu_unprepare(lpm_cpu, idx, false);
-	return 0;
+	return ret;
 }
 
 static const struct platform_suspend_ops lpm_suspend_ops = {
-- 
2.49.0

